ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToClient.hs

```diff
@@ -303,7 +303,7 @@ nullTracers = Tracers {
 showTracers :: ( Show peer
                , Show (GenTx blk)
                , Show (ApplyTxErr blk)
-               , ShowQuery (Query blk)
```

Comment:
Hmm. This seems a bit overealous, since line 280 above still has:

```
tStateQueryTracer   :: f (TraceLabelPeer peer (TraceSendRecv (LocalStateQuery blk (Point blk) (Query blk))))
```

We probably want to leave the constraints on `Query` eg so that when we do add new top-level-only-queries this constraint need not change. Does that make sense to you?

I'm guessing this is the result of a find-and-replace etc, so there are probably other signatures and eg other classes that would require the same treatment.

Reply:
IIUC, John Ky has written some of these instances already on his PR #3110

Reply:
I think I did that originally but it results in a warning:

```
src/Ouroboros/Consensus/Network/NodeToClient.hs:303:16: warning: [-Wsimplifiable-class-constraints]
    • The constraint ‘ShowQuery (Query blk)’ matches
        instance ShowQuery (BlockQuery blk) => ShowQuery (Query blk)
          -- Defined at src/Ouroboros/Consensus/Ledger/Query.hs:50:10
      This makes type inference for inner bindings fragile;
        either use MonoLocalBinds, or simplify it using the instance
    • In the type signature:
        showTracers :: (Show peer,
                        Show (GenTx blk),
                        Show (ApplyTxErr blk),
                        ShowQuery (Query blk),
                        HasHeader blk) =>
                       Tracer m String -> Tracers m peer blk e
    |
303 | showTracers :: ( Show peer
    |                ^^^^^^^^^^^...
```

I could suppress the warning, but it seems simpler to just wait till the new top level query is added. At that point I think GHC will remind me to adjust these constraints. How does that sound?

Reply:
Ah, I see! Harumph.

Yes, I think the extra git commit churn on these lines is probably the right choice instead of adding the warning suppressor to the whole module. (If pragmas could be specified on a per top-level declaration basis I might make the other choice :/)